tests/test_packages/test_connections/test_p2p_libp2p/test_local_dht.py

```diff
@@ -0,0 +1,299 @@
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This test module contains integration tests for P2PLibp2p connection."""
+
+import itertools
+import json
+import os
+
+import pytest
+
+from aea.test_tools.test_cases import AEATestCaseMany
+
+from packages.valory.connections import p2p_libp2p, p2p_libp2p_client
+from packages.valory.connections.p2p_libp2p.connection import (
+    PUBLIC_ID as P2P_CONNECTION_PUBLIC_ID,
+)
+from packages.valory.connections.p2p_libp2p_client.connection import (
+    PUBLIC_ID as P2P_CLIENT_CONNECTION_PUBLIC_ID,
+)
+
+from tests.conftest import DEFAULT_LEDGER, UseACNNode, UseACNWithBootstrappedEntryNodes
+from tests.test_packages.test_connections.test_p2p_libp2p.base import (
+    BaseP2PLibp2pTest,
+    LIBP2P_LEDGER,
+    libp2p_log_on_failure_all,
+    load_client_connection_yaml_config,
+    make_cert_request,
+    ports,
+)
+
+
+genesis_nodes = load_client_connection_yaml_config()["nodes"]
+
+PUBLIC_DHT_MADDRS = [
+    "/dns4/0.0.0.0/tcp/10000/p2p/16Uiu2HAmMC2tJMRaRTeWSESv8mArbq6jipJCD4adSBcBLsbc7cSL"
+]
+PUBLIC_DHT_DELEGATE_URIS = ["localhost:11000"]
+PUBLIC_DHT_PUBLIC_KEYS = [
+    "037ed15dcee3a317e590cbdd28768ad8e2d29960b3e5d4eccca14bc94f83747f09"
+]
+
+AEA_DEFAULT_LAUNCH_TIMEOUT = 30
+AEA_LIBP2P_LAUNCH_TIMEOUT = 30
+
+p2p_libp2p_path = f"vendor.{p2p_libp2p.__name__.split('.', 1)[-1]}"
+p2p_libp2p_client_path = f"vendor.{p2p_libp2p_client.__name__.split('.', 1)[-1]}"
+
+
+@pytest.fixture
+def maddrs(request):
+    """Fixture for multi addresses."""
+    return request.param
+
+
+@pytest.fixture
+def delegate_uris_public_keys(request):
+    """Fixture for delegate uris and public keys."""
+    return request.param
+
+
+@pytest.mark.integration
+@libp2p_log_on_failure_all
+class TestLibp2pConnectionDHTDelegate(
+    BaseP2PLibp2pTest, UseACNWithBootstrappedEntryNodes
+):
+    """Test that local DHTs delegate service is working properly"""
+
+    def setup(self):
+        """Set up test"""
+        uri = "localhost:11000"
+        public_key = (
+            "037ed15dcee3a317e590cbdd28768ad8e2d29960b3e5d4eccca14bc94f83747f09"
+        )
+        self.make_client_connection(uri=uri, peer_public_key=public_key)
+
+    def test_connectivity(self):
+        """Test connectivity."""
+        assert self.all_connected
```

Comment:
this is just a temporary test class I'm using while in the process of trying to replace `UseACNNode` with `UseACNWithBootstrappedEntryNodes`